tests/forte/data/ontology/ontology_code_generator_test.py

```diff
@@ -101,13 +94,51 @@ def test_dry_run_false(self):
         temp_dir = tempfile.mkdtemp()
         json_file_path = os.path.join(
             self.spec_dir, "example_import_ontology.json")
-        temp_filename = get_temp_filename(json_file_path, temp_dir)
-        self.generator.generate(temp_filename, temp_dir, False)
+        temp_filename = _get_temp_filename(json_file_path, temp_dir)
+        self.generator.generate(temp_filename, temp_dir, is_dry_run=False)
         folder_path = temp_dir
         for name in ["ft", "onto", "example_import_ontology.py"]:
             self.assertTrue(name in os.listdir(folder_path))
             folder_path = os.path.join(folder_path, name)
 
+    def test_include_and_exclude_init(self):
+        temp_dir = tempfile.mkdtemp()
+        json_file_path = os.path.join(
+            self.spec_dir, "example_import_ontology.json")
+        temp_filename = _get_temp_filename(json_file_path, temp_dir)
+
+        # Test with include_init = True
+        folder_path = self.generator.generate(temp_filename, temp_dir,
+                                              is_dry_run=False,
+                                              include_init=True)
+        gen_files = sorted(utils.get_generated_files_in_dir(folder_path))
+
+        # Assert the generated python files
+        exp_file_path = ['ft/__init__',
+                         'ft/onto/__init__',
+                         'ft/onto/example_import_ontology']
+        exp_files = sorted([f"{os.path.join(folder_path, file)}.py"
+                           for file in exp_file_path])
+
+        self.assertCountEqual(gen_files, exp_files)
+
+        # Now, corrupt one of the init files
+        corrupted_path = os.path.join(folder_path, 'ft/__init__.py')
+        with open(corrupted_path, 'w') as f:
+            f.write('# ***corrupted file***\n')
+
+        # Re-generate using include_init = False
+        self.generator = OntologyCodeGenerator()
+        folder_path = self.generator.generate(temp_filename, folder_path,
+                                              is_dry_run=False,
+                                              include_init=False)
+        gen_files = sorted(utils.get_generated_files_in_dir(folder_path))
+
+        # Assert the generated python files after removing the corrupted file
+        # which should not have been regenerated
+        exp_files = [file for file in exp_files if file != corrupted_path]
+        self.assertCountEqual(gen_files, exp_files)
```

Comment:
it is more accurate to compare the list content (after sorted)

Reply:
Done.